### ISSUE TITLE
treebank: option to show all inflections, highlighting the one from the tree

### ISSUE THREAD

**Issue by balmas** [alpheios-project/alpheios-core#540]
What Alpheios does is disambiguate the lexemes identified by the parser with the lexeme in the treebank.

For the lexemes themselves (i.e. the distinct lemma entries, as far as we can determine them) it identifies which lexeme is the one chosen by the treebank with the little red triangle, but still shows the other lexemes.
 
However, for the inflections within the disambiguated lexeme, it filters the inflections to show *only* the one that was chosen by the treebank.

@vgorman1 requests that we have the possibility to show the various possible inflections and mark the one in the tree with a similar triangle.

**Comment by balmas** [comment]
this is fixed in Alpheios Components 3.3.1-qa.20210106574

For now, the default has been changed to always display all inflections, and to highlight the selected inflection with the same icons we use for the lemma:

![c6f057a5-bbc9-4357-8cde-7030030eb139](https://user-images.githubusercontent.com/644870/103824007-1d52e400-5041-11eb-8952-7acd4c8ed0b4.png)

It's not perfect -- when there are more than 1 possible inflection identified by morpheus, those are displayed below the selected one from the tree, including (again) the one from the tree.  Ideally, I would like to dedupe the inflection from the tree out of the list of the inflections identified by morpheus, but that is too big of a change for me to make right now.

Still to be determined is whether we want this changed default behavior to apply to the treebanked texts at texts.alpheios.net -- need to verify that with @abrasax - but it's ready to test otherwise.

You can use treebanked texts at texts-test.alpheios.net to test as well as the treebank-test page at https://alpheios-misc-dev.s3.us-east-2.amazonaws.com/treebank-test-page/test.html

**Comment by monzug** [comment]
I really like this enhancement!!!

**Comment by monzug** [comment]
in the case of nullis (first sentence of https://texts-test.alpheios.net/text/urn:cts:latinLit:phi0620.phi001.alpheios-text-lat1/passage/1.1), we are adding again the inflection. see screenshot
as Bridget said: It's not perfect -- when there are more than 1 possible inflection identified by morpheus, those are displayed below the selected one from the tree, including (again) the one from the tree. Ideally, I would like to dedupe the inflection from the tree out of the list of the inflections identified by morpheus, but that is too big of a change for me to make right now.

in texts-test version vs the live one
![nullis-cynthia](https://user-images.githubusercontent.com/41396793/104196764-2a315800-53f2-11eb-8b99-01581203b853.JPG)

**Comment by monzug** [comment]
In the case of fugiendo (line 9), we are adding too much, I believe.
- lookup for fugiendo returns  a verb participle fut. pass. -
- the treebank of  fugiendo in texts.alpheios.net  says only verb participle.
- the treebank of fugiendo in https://texts-test.alpheios.net/text/urn:cts:latinLit:phi0620.phi001.alpheios-text-lat1/passage/1.1   adds   a  pres. pass.  - see screenshot

![fugiendo](https://user-images.githubusercontent.com/41396793/104200394-9a41dd00-53f6-11eb-9db9-ea90257827e8.JPG)

**Comment by balmas** [comment]
Well our requirements right now call us to consider the treebank data, which is manually annotated to be considered more correct than the parser output. So the code is doing the expected thing here -- it's adding the present passive inflection provided by the treebank to the form, and saying that's the "correct" one.

**Comment by monzug** [comment]
absolutely! code is right, treebank wrong. However, would it be possible to
add an extra layer that checks if form is in morphology? if not, drop it or
add a comment. I do not know. just thinking. I do not even know how
frequently we could have such a scenario.

**Comment by balmas** [comment]
When you say "form is in morphology" do you mean, check that the form is one that was returned by the morphology parser (in this case whitaker)? 

If so, the problem is that we don't have a way to differentiate which source is right -- i.e. the parser or the treebank.  Ultimately this is why we need active annotation support in Alpheios, which is the subject of our next major release.  (see lengthy ongoing discussions of the design to support that at https://github.com/alpheios-project/documentation/issues/40 )k

**Comment by vgorman1** [comment]
And there are definitely times when the morphology yielded is not a possible annotation.  My most frequent example is δεῖ, which is always analyzed as 3s imperfect ind act, when it is actually present tense and ἔδει is the imperfect [as per the LSJ].  Clearly it was just entered wrong at some point.  There are also real doozies of misspellings in Morpheus now and then.

So, as we all agree, the ideal is to mark the tree annotation but include the other possibilities.  You might also be able to use form and frequency data from various users somehow

Vanessa B. Gorman
Professor of Ancient History
Department of History
619 Oldfather Hall
University of Nebraska-Lincoln
Lincoln, NE 68588-0327
https://vgorman1.github.io/

**Comment by balmas** [comment]
another interesting point here, which @rgorman helped clarify -- the treebank reports this as mood=gerundive. The Whitake parser used by Alpheios reports all Gerunds as verb participles (see alpheios-project/morphsvc#11).  

I think perhaps we should make a change to the Alpheios treebank adapter to consider a Latin verb with mood=gerundive as being the same as a verb participle. That way we will at least be comparing apples to apples. We'd still have a disconnect here, because the tense in the treebank is present. But we would at least be matching the part of speech.

**Comment by monzug** [comment]
gerundive verbs have been hunting me for ever! issue #608 is very welcome.

**Comment by balmas** [comment]
Wrt to https://github.com/alpheios-project/alpheios-core/issues/540#issuecomment-757999170 there are actually two issues here, one which I didn't see at first

(1)
Morphology Service says the lemma has inflection A and inflection B
Treebank says it is inflection B
We should show inflection B only once in the popup.


Originally I thought that was what was going on with nullis. That has to wait for the changes we're making to support annotations.  However, looking more closely at the output, I realize that is actually a different scenario:

(2) 
Morphology service says inflection A m,f,n
Treebank service says inflection A f
We should recognize that Inflection A f is not different than Inflection A m, f n

This scenario is more similar to #608 and #609.   Will enter a new issue for it.

**Comment by monzug** [comment]
verified. All comments have been addressed in separate issues.
